### PR TITLE
One python version per container

### PR DESCRIPTION
**Pull request by divyegala**

### CODE REVIEW

File: ci/compute-mx.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -euo pipefail
+
+yq -o json axis.yaml | jq -c 'include "ci/compute-mx"; compute_mx(.)'
+
+# MATRIX=$(yq -o json '.' axis.yaml | jq -c)
+# echo "MATRIX=${MATRIX}" | tee --append ${GITHUB_OUTPUT:-/dev/null}
```
Comment (ajschmidt8) [inline ci/compute-mx.sh:8]: these comments can probably be removed.

File: axis.yaml [hunk 2]
```diff
@@ -1,29 +1,27 @@
...
-  - "rapidsai/manylinux_v2_2_31:cuda-devel-12.0.1-ubuntu20.04"
-RUNNER_ARCH:
-  - "amd64"
-  - "arm64"
+CUDA_VER:
+  - "11.8.0"
+  - "12.0.1"
+PYTHON_VER:
+  - "3.9.16"
+  - "3.10.9"
```
Comment (ajschmidt8) [inline axis.yaml:6]: To follow the conda image conventions, are we able to omit the patch version here?

It looks like the only place these values are used are in the `pyenv global ${PYTHON_VER}` command in the Dockerfiles.

Does that command work with just a major minor version?

File: .github/workflows/prs.yml
[the remark this answers is not in the document]
Comment (divyegala) [reply .github/workflows/prs.yml:1]: Should I also rename `merge.yml` to `build.yml` like other RAPIDS repos then?

File: axis.yaml (hunk 2 above)
Comment (divyegala) [reply axis.yaml:6]: Good question, let me try without and see what happens

File: build.sh
[the remark this answers is not in the document]
Comment (divyegala) [reply build.sh:1]: Yep, I will also remove `workflows-disable` at that point

File: axis.yaml (hunk 2 above)
Comment (divyegala) [reply axis.yaml]: Okay, this worked just fine

**Comment by divyegala** [comment]
@ajschmidt8 this is ready for a re-review

File: ci/compute-mx.jq [hunk 3]
```diff
@@ -0,0 +1,36 @@
+def compute_arch($x):
+  ["amd64"] |
+  if
+    $x.LINUX_VER == "ubuntu20.04"
+  then
+    . + ["arm64"]
+  else
+    .
+  end |
+  $x + {ARCHES: .};
```
Comment (ajschmidt8) [inline ci/compute-mx.jq:10]:
```suggestion
def compute_arch($x):
  ["amd64"] |
  if
    ["ubuntu18.04", "centos7"] | any(index($x.LINUX_VER))
  then
    .
  else
    . + ["arm64"]
  end |
  $x + {ARCHES: .};
```

I think we should replace the `== "ubuntu20.04"`  with this suggested logic.

This way we don't have to explicitly add any new statements here when we add newer OSes, which will presumably have ARM support.

File: .github/workflows/build-and-publish-images.yml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    with:
+      ARCHES: ${{ toJSON(matrix.ARCHES) }}
+      CUDA_VER: ${{ matrix.CUDA_VER }}
+      LINUX_VER: ${{ matrix.LINUX_VER }}
+      PYTHON_VER: ${{ matrix.PYTHON_VER }}
+      IMAGE_REPO: ${{ matrix.IMAGE_REPO }}
+      TAG: rapidsai/${{ matrix.IMAGE_REPO }}:cuda${{ matrix.CUDA_VER }}-${{ matrix.LINUX_VER }}-py${{ matrix.PYTHON_VER }}
+      BUILD_TYPE: ${{ inputs.build_type }}
+  build-multiarch-manifest:
+    # if: inputs.build_type == 'branch'
```
Comment (ajschmidt8) [inline .github/workflows/build-and-publish-images.yml:43]:
```suggestion
    if: inputs.build_type == 'branch'
```

File: .github/workflows/build-and-publish-images.yml [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+          if [[
+            "${LATEST_UBUNTU_VER}" == "${{ matrix.LINUX_VER }}" &&
+            "${LATEST_CUDA_VER}" == "${{ matrix.CUDA_VER }}" &&
+            "${LATEST_PYTHON_VER}" == "${{ matrix.PYTHON_VER }}"
+          ]]; then
+            docker manifest create rapidsai/${{ matrix.IMAGE_REPO }}:latest ${source_tags[@]}
+            docker manifest push rapidsai/${{ matrix.IMAGE_REPO }}:latest
+          fi
+  delete-temp-images:
+    # if: inputs.build_type == 'branch'
```
Comment (ajschmidt8) [inline .github/workflows/build-and-publish-images.yml:79]:
```suggestion
    if: inputs.build_type == 'branch'
```

File: .github/workflows/build-image.yml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            echo "MANYLINUX_VER=manylinux_2_17" >> $GITHUB_OUTPUT
+          fi
+      - name: Build image
+        if: 
+        uses: docker/build-push-action@v4
+        with:
+          context: ./${{ inputs.IMAGE_REPO }}
+          file: ./${{ inputs.IMAGE_REPO }}/Dockerfile
+          # push: ${{ inputs.BUILD_TYPE == 'branch' }}
+          push: true
```
Comment (ajschmidt8) [inline .github/workflows/build-image.yml:72]:
```suggestion
          push: ${{ inputs.BUILD_TYPE == 'branch' }}
```

File: .github/workflows/build-and-publish-images.yml (hunk 4 above)
Comment (divyegala) [reply .github/workflows/build-and-publish-images.yml:43]: I'll come back to this once we have a branch upstream apart from `main` that I can use to push images. Right now, this is enabling me to build images and test them in https://github.com/rapidsai/raft/pull/1546

**Review by vyasr — CHANGES_REQUESTED** [review]
Some initial requests:
- Move the deletion of the disabled workflows to a separate PR
- Rename workflows to match the workflows used in our other repos (pr/build/test) assuming they follow the same logic (I don't think we need the equivalent of a "test" workflow nightly)
- Let's leave the `ci-wheel` images named something with `manylinux` in them. Perhaps `manylinux_cuda`? I still want that use case to be clear.
- Can we separate change to the manylinux containers from changes to the citestwheel images (separate PRs)?

**Comment by ajschmidt8** [comment]
> * Let's leave the `ci-wheel` images named something with `manylinux` in them. Perhaps `manylinux_cuda`? I still want that use case to be clear.

Hmm. I was hoping to keep these as `ci-wheel` in anticipation of these images being merged into the https://github.com/rapidsai/ci-imgs/ repository.

Once that happens, I assumed we'd have two images in that repository: `ci-conda` and `ci-wheel`.

Any thoughts on this approach?

File: .github/workflows/push.yml [hunk 7]
```diff
@@ -1,20 +1,17 @@
-name: Build and push RAPIDS pip wheel CI images
+name: publish
 
 on:
   push:
     branches:
       - "main"
```
Comment (ajschmidt8) [inline .github/workflows/push.yml:6]:
```suggestion
      - "imgs-v2"
```

**Review by vyasr — APPROVED** [review]
Images look good. Didn't review the rest.

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**